src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+		transformOptions.setOutputURIResolver(outputURIResolver);
+		
+		TransformStreamProvider baseStreamProvider = transformerService.getTransformStreamProvider(baseItemId, transformOptions);
+		addToPatchset(patchset, outputPath.append(baseItemId.getRelPath().getName()), baseStreamProvider, overwrite, props);
+		
+		Set<String> resultDocsHrefs = outputURIResolver.getResultDocumentHrefs();
+		for (String relpath: resultDocsHrefs) {
+			if (relpath.startsWith("/")) {
+				throw new IllegalArgumentException("Relative href must not start with slash: " + relpath);
+			}
+			
+			XmlSourceDocumentS9api resultDocument = outputURIResolver.getResultDocument(relpath);
+			TransformStreamProvider streamProvider = transformerIdentity.getTransformStreamProvider(resultDocument, null);
+			
+			CmsItemPath path = outputPath.append(Arrays.asList(relpath.split("/")));
+			addToPatchset(patchset, path, streamProvider, overwrite, props);
+		}
+		
+		commit.run(patchset);
+	}
+	
+	private void addToPatchset(CmsPatchset patchset, CmsItemPath relPath, TransformStreamProvider streamProvider, boolean overwrite, CmsItemPropertiesMap properties) {
+		try {
+			
+			final InputStream transformStream = checkStreamIsNotEmpty(streamProvider.get());
+			boolean pathExists = pathExists(patchset.getRepository(), relPath);
+			if (!pathExists) {
+				logger.debug("No file at path: '{}' will add new file.", relPath);
+				FileAdd fileAdd = new FileAdd(relPath, transformStream);
+				fileAdd.setPropertyChange(properties);
+				patchset.add(fileAdd);
+				addFolderExists(patchset, relPath.getParent());
+			} else if (overwrite){
+				logger.debug("Overwrite is allowed, existing file at path '{}' will be modified.", relPath.getPath());
+				CmsItemId itemId = new CmsItemIdArg(patchset.getRepository(), relPath);
+				CmsItemLockCollection locks = commit.lock(TRANSFORM_LOCK_COMMENT, patchset.getBaseRevision(), itemId.getRelPath());
+				if (locks != null && locks.getSingle() == null) {
+					throw new IllegalStateException("Unable to retrieve the lock token after locking " + itemId);
+				}
+				patchset.addLock(locks.getSingle());
+				FileModificationLocked fileMod = new FileModificationLocked(relPath, transformStream);
+				fileMod.setPropertyChange(properties);
+				patchset.add(fileMod);
+			} else {
+				throw new IllegalStateException("Item already exists, config prohibiting overwrite of existing items.");
+			}
+			
+		} catch (IOException e) {
+			throw new RuntimeException("Failed to read stream from transform.", e);
+		} catch (EmptyStreamException e) {
+			logger.warn("Transform of item at path: '{}'  resulted in empty document, will be discarded.", relPath);
+		}
+	}
+	
+	private CmsItemPath getOutputPath(CmsItemId itemId, String output) {
+		
+		CmsItemPath pathResult;
+		
+		if (output == null || output.trim().isEmpty()) {
+			pathResult = new CmsItemIdArg(itemId.getRepository(), itemId.getRelPath().getParent()).getRelPath();
```

Comment:
See javadoc of CmsItemIdArd constructor.

Reply:
Changed all occurrences  of CmsItemIdArg to: repo.getItemId().withRelpath(....);